src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.model.socialmedia.SocialMedia;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s information (including name, phone, email, social media handles, and tags)
+ * matches any of the keywords given.
+ */
+public class PersonContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public PersonContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (keywords.stream()
```

Comment:
Is it possible to abstract out the conditional statements into separate methods such that each one tests for name, phone, etc?